### PR TITLE
New infowindows

### PR DESCRIPTION
**Pull request by javierarce**
This PR solves several tasks described in the original issue: https://github.com/CartoDB/cartodb.js/issues/899

There are two pending related tickets here: 
https://github.com/CartoDB/cartodb.js/issues/932
https://github.com/CartoDB/cartodb.js/issues/931

I propose to merge this and address them separately.

### CODE REVIEW

File: src/geo/ui/infowindow.js [hunk 1]
```diff
@@ -369,8 +396,8 @@ var Infowindow = View.extend({
    */
   _isValidURL: function(url) {
     if (url) {
-      var urlPattern = /^(http|ftp|https):\/\/[\w-]+(\.[\w-]+)+([\w.,@?^=%&amp;:\/~+#-|]*[\w@?^=%&amp;\/~+#-])?$/
-      return String(url).match(urlPattern) != null ? true : false;
+      var urlPattern = /^(http|ftp|https):\/\/[\w-]+(\.[\w-]+)+([\w.,@?^=%&amp;:\/~+#-|]*[\w@?^=%&amp;\/~+#-])?$/;
```
Comment (javierarce) [inline src/geo/ui/infowindow.js]: added ; (not for very long)

**Comment by javierarce** [comment]
cc: @xavijam

**Comment by xavijam** [comment]
:clap: reviewing now. By the way, those tickets in description are the same :dancer: :D.

**Comment by javierarce** [comment]
Oops, changed. It was a very important ticket.

File: src/geo/ui/infowindow.js [hunk 2]
```diff
@@ -21,6 +22,11 @@ var util = require('../../core/util');
 
 var Infowindow = View.extend({
 
+  options: {
+    imageTransitionSpeed: 300,
+    hookHeight: 16
+  },
+
   className: "cartodb-infowindow",
```
Comment (xavijam) [inline src/geo/ui/infowindow.js]: Do you think this class should be reviewed?

File: src/geo/ui/infowindow.js [hunk 3]
```diff
@@ -307,17 +318,42 @@ var Infowindow = View.extend({
     return false;
   },
 
+  _loadImageHook: function(url) {
```
Comment (xavijam) [inline src/geo/ui/infowindow.js]: :clap:

**Comment by xavijam** [comment]
Probably this is a historic moment: could we remove jScrollPane library??? 
```redoble de tambor```

File: src/geo/ui/infowindow.js [hunk 4]
```diff
@@ -291,7 +302,7 @@ var Infowindow = View.extend({
    *  Does header contain cover?
    */
   _containsCover: function() {
-    return this.$el.find(".cartodb-popup.header").attr("data-cover") ? true : false;
+    return this.$el.find(".js-infowindow").attr("data-cover") ? true : false;
```
Comment (xavijam) [inline src/geo/ui/infowindow.js]: this.$('.js-infowindow') works too.

File: src/geo/ui/infowindow.js [hunk 5]
```diff
@@ -327,39 +363,39 @@ var Infowindow = View.extend({
...
         }
       }
-
+      
       $img.css(styles);
-      $img.fadeIn(300);
+      $cover.css({ height: h - self.options.hookHeight });
+      $img.fadeIn(self.options.imageTransitionSpeed);
+
+      self._loadImageHook(url);
```
Comment (xavijam) [inline src/geo/ui/infowindow.js]: Shouldn't we load/set hook image when there is only image + title?
cc @piensaenpixel

File: src/geo/ui/infowindow.js (hunk 1 above)
Comment (xavijam) [inline src/geo/ui/infowindow.js]: :joy:

File: src/geo/ui/infowindow.js (hunk 2 above)
Comment (javierarce) [inline src/geo/ui/infowindow.js]: True

**Comment by javierarce** [comment]
> Probably this is a historic moment: could we remove jScrollPane library??? 

I thought that, but this scroll is being used in the [mobile view](https://github.com/CartoDB/cartodb.js/blob/v4-module-export/src/geo/ui/mobile.js)… I created a ticket for that [here](https://github.com/CartoDB/cartodb.js/issues/934)

**Comment by xavijam** [comment]
Ohhh, mobile view, ok!

File: src/geo/ui/infowindow.js [hunk 6]
```diff
@@ -307,17 +318,42 @@ var Infowindow = View.extend({
...
 
     var self = this;
-    var $cover = this.$(".cover");
+
+    var $cover = this.$(".js-cover");
     var $img = $cover.find("img");
-    var $shadow = this.$(".shadow");
     var url = this._getCoverURL();
 
     if ($img.length > 0) {
```
Comment (xavijam) [inline src/geo/ui/infowindow.js]: If there is already an image defined in the header cover, we should add the class ```CDB-infowindow-media-item``` to that image and we should run ```this._loadImageHook(url);``` before the ```return;```.

**Comment by xavijam** [comment]
Last two things:

- CDB-hook magic should not be applied if the infowindow has any content outside header cover.
- Check what clip-path thing is doing because some svgs are being added to the body when an image infowindow is displayed.

**Comment by javierarce** [comment]
I've created issues for two pending tasks, so merging this branch.

**Comment by xavijam** [comment]
:clap:

**Pull request merged**